text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,1026 @@
+   * @default no vpcs are associated
+   */
+  readonly vpcs?: ec2.IVpc[] | undefined;
+  /**
+   * An access mode provides 'pre-canned' conditions for the policy
+   * @default no conditions
+   */
+  readonly accessmode?: ServiceNetworkAccessMode | undefined;
+}
+
+```
+`ServiceNetwork` will implement statics in addition to those in IServiceNetwork, to allow the import of the servicenetwork in other stacks ( potentially cross account ). `fromName` provides a lookup to obtain the Id, using a custom resource.  This provides a way to pass a concrete value between cross account stacks.  
+
+```typescript
+public static fromId(scope: constructs.Construct, id: string, props: ImportedServiceNetworkProps ): IServiceNetwork {
+    return new ImportedServiceNetwork(scope, id, props);
+}
+
+public static fromName(scope: constructs.Construct, id: string, serviceNetworkName: string ): IServiceNetwork {
+    return new ImportedServiceNetwork(scope, id, { serviceNetworkName: serviceNetworkName });
+  }
+```
+
+`ServiceNetwork` will implement methods for a variety of tasks;
+
+
+```typescript
+// adds an iam statement to the the authpolicy. Use this to create policy that is bespoke to this application 
+addStatementToAuthPolicy(statement: iam.PolicyStatement)
+```
+
+```typescript
+// applies the AuthPolicy that has been created, to the serviceNetwork. 
+applyAuthPolicyToServiceNetwork()
+```
+
+```typescript
+// creates and associates a logging subscription to provide visisiblity for the lattice service
+addloggingDestination(props: AddloggingDestinationProps)
+```
+
+```typescript
+// share the service network using RAM
+ public share(props: ShareServiceNetworkProps)
+```
+
+### ServiceNetworkAssociation
+
+Creates a Cloudformation `AWS::VpcLattice::ServiceNetworkServiceAssociation`
+Associates a service with a service network.
+`ServiceNetworkAssociation` extends core.Resource and consumes ServiceNetworkAssociationProps
+Consider using `.associateWithServiceNetwork` method on `Service` in preference.
+
+
+```typescript
+export interface ServiceNetworkAssociationProps {
+  /**
+   * lattice Service
+   */
+  readonly serviceNetwork: IServiceNetwork;
+  /**
+   * Lattice ServiceId
+   */
+  readonly serviceId: string;
+}
+```
+
+
+
+
+### Service
+
+Creates a cloudformation `AWS::VpcLattice::Service`
+
+A service within VPC Lattice is an independently deployable unit of software that delivers a specific task or function.  A service has listeners that use rules, that you configure to route traffic to your targets. Targets can be EC2 instances, IP addresses, serverless Lambda functions, Application Load Balancers, or Kubernetes Pods.  The following diagram shows the key components of a typical service within VPC Lattice.
+
+
+![service](https://docs.aws.amazon.com/images/vpc-lattice/latest/ug/images/service.png)
+
+
+`Service` extends `core.Resource` and implments `IService` 
+
+```typescript
+export interface IService extends core.IResource {
```

Comment:
We probably want this to also extend `IGrantable` so that we can do `grantAccess(service)`

Reply:
We dont' need to grant access to the service,    access to the service is only via that authpolicy.